src/alexapi/player.py

```diff
@@ -0,0 +1,302 @@
+
+
+	def queue_and_play(self, file, offset=0, overRideVolume=0):
+		global media_playing, media_paused
+		#TODO: add volumue override
+
+		if file:
+			self.addItem(file)
+
+		while True:
+			if media_playing is False and media_paused is False:
+				media_paused = False
+
+				if self.getItemCount() == 0: # No more items in queue. Quit loop
+					#TODO: Do stuff
+					break
+
+				media_playing = True
+
+				self.player = self.instance.media_player_new()
+				media = self.instance.media_new(self.getNextItem())
+				self.player.set_media(media)
+				events = media.event_manager()
+				events.event_attach(vlc.EventType.MediaStateChanged, self.media_player_callback, self.player)
+
+				if (overRideVolume == 0):
+					self.player.audio_set_volume(pstate.currVolume)
+				else:
+					self.player.audio_set_volume(overRideVolume)
+
+				if debug: print "{}Queue playing:{} {}".format(bcolors.OKBLUE, bcolors.ENDC, self.getCurrentItem())
+				self.player.play()
+				time.sleep(.1) # Allow time for state_callback to run
+
+
+				while media_playing:
+					time.sleep(.5) # Prevent 100% CPU untilzation
+					continue
+
+				#TODO: Do I still need this? 
+				#media_playing = False
+
+			time.sleep(1) # Prevent 100% CPU untilzation
+
+
+	def stop(self):
+		self.q.clear()
+		self.player.stop()
+
+	def resume(self):
+		queue_and_play(self)
+
+	def is_playing(self):
+		global media_playing
+		return media_playing
+
+
+	def media_player_callback(self, event, player):
+		global media_playing, alexa_playback_progress_report_request, alexa_getnextitem
+
+		state = player.get_state()
+
+		if debug: print("{}Media Player State:{} {}".format(bcolors.OKGREEN, bcolors.ENDC, state))
+		if state == 3:		#Playing
+			media_playing = True
+			media_paused = False
+
+		elif state == 4:	#Paused
+			media_playing = False
+			media_paused = True
+
+		elif state == 5:	#Stopped
+			media_playing = False
+
+		elif state == 6:	#Ended
+			media_playing = False
+			if pstate.streamid != "":
+				rThread = threading.Thread(target=alexa_playback_progress_report_request, args=("FINISHED", "IDLE", pstate.streamid))
+				rThread.start()
+				pstate.streamid = ""
+
+			if pstate.streamurl != "":
+				pThread = threading.Thread(target=play, args=(pstate.streamurl,))
+				pThread.start()
+				pstate.streamurl = ""
+
+			elif pstate.nav_token != "":
+				gThread = threading.Thread(target=alexa_getnextitem, args=(pstate.nav_token,))
+				gThread.start()
+
+		elif state == 7:	#Error
+			media_playing = False
+
+
+# Initialize players
+alexa_voice_response_player = AlexaVoiceResponsePlayer()
+media_player = MediaPlayer()
+
+
+"""
+Setup
+"""
+def setup(playback_progress_report_request, getnextitem, tplaylist):	#TODO:I'm sure there is a better way
+	global alexa_playback_progress_report_request, alexa_getnextitem, tuneinplaylist
+
+	tuneinplaylist = tplaylist
+	alexa_playback_progress_report_request=playback_progress_report_request
+	alexa_getnextitem = getnextitem
+
+"""
```

Comment:
This line and down (I don't know how to select multiple lines):
Get rid of these encapsulations. Just create objects in `main.py` that call the stuff directly.